src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -808,37 +809,36 @@ public unsafe void CopyTo(Array array, int index)
                 if (m_length < BitsPerInt32)
                     goto LessThan32;
 
-                // The mask used when shuffling a single int into Vector128/256.
-                // On little endian machines, the lower 8 bits of int belong in the first byte, next lower 8 in the second and so on.
-                // We place the bytes that contain the bits to its respective byte so that we can mask out only the relevant bits later.
-                Vector128<byte> lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
-                Vector128<byte> upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
-
-                if (Vector256.IsHardwareAccelerated)
+                if (Avx2.IsSupported)
                 {
-                    Vector256<byte> shuffleMask = Vector256.Create(lowerShuffleMask_CopyToBoolArray, upperShuffleMask_CopyToBoolArray);
+                    // The mask used when shuffling a single int into Vector128/256.
+                    // On little endian machines, the lower 8 bits of int belong in the first byte, next lower 8 in the second and so on.
+                    // We place the bytes that contain the bits to its respective byte so that we can mask out only the relevant bits later.
+                    Vector128<byte> lowerShuffleMask = Vector128.Create(0, 0x01010101_01010101).AsByte();
+                    Vector128<byte> upperShuffleMask = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
+
+                    Vector256<byte> shuffleMask = Vector256.Create(lowerShuffleMask, upperShuffleMask);
                     Vector256<byte> bitMask = Vector256.Create(0x80402010_08040201).AsByte();
                     Vector256<byte> ones = Vector256.Create((byte)1);
 
-                    ref byte destination = ref Unsafe.As<bool, byte>(ref Unsafe.Add<bool>(ref MemoryMarshal.GetArrayDataReference<bool>(boolArray), index));
-
-                    for (; (i + Vector256ByteCount) <= (uint)m_length; i += Vector256ByteCount)
+                    fixed (bool* destination = &boolArray[index])
```

Comment:
It would be better to use `Vector256` APIs for everything except `shuffle` if `shuffle` has an issue (but local tests show it shouldn't).